src/caret_analyze/common/type_check_decorator.py

```diff
@@ -174,7 +174,8 @@ def _custom_wrapper(*args, **kwargs):
                 return validate_arguments_wrapper(*args, **kwargs)
             except ValidationError as e:
                 expected_types = _get_expected_types(e)
-                loc_tuple = e.errors()[0]['loc']
+                loc_tuple = e.errors()[0]['loc'] if len(e.errors()) == 1\
+                    else e.errors()[0]['loc'][:-1]
```

Comment:
Can't we simply get the index=0 data here?
If it does not work as intended, forget the above comment.

e.error_count() is provided as a way to get the number of errors.

Reply:
docs for error_count() is [here](https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core._pydantic_core.ValidationError.error_count)

Reply:
L178 was a code added in L157 (before the change) to determine if it is an Iterable type case.

In fact, there is a better code to determine if it is an Iterable type case, so the code has been updated ([0a77f93](https://github.com/tier4/CARET_analyze/pull/311/commits/0a77f935c173c6a71ff5499b3a0e1dddf977b6cc)).

Reply:
Thanks for the correction.
I think the code is now more readable.